File: swarm_input/calculator2.rb
class ScientificCalculator
  def square_root(n); Math.sqrt(n); end
  def power(base, exp); base ** exp; end
  def log(n); Math.log(n); end
  def factorial(n); n <= 1 ? 1 : n * factorial(n - 1); end
end

calc = ScientificCalculator.new
puts calc.square_root(16)
puts calc.power(2, 10)
puts calc.factorial(5)

File: swarm_input/calculator2.py
import math

class ScientificCalculator:
    def square_root(self, n):
        return math.sqrt(n)

    def power(self, base, exp):
        return base ** exp

    def log(self, n):
        return math.log(n)

    def factorial(self, n):
        if n <= 1:
            return 1
        else:
            return n * self.factorial(n - 1)

calc = ScientificCalculator()
print(calc.square_root(16))
print(calc.power(2, 10))
print(calc.factorial(5))
